src/rendering/RenderUtil.cc

```diff
@@ -235,6 +242,30 @@ rendering::ScenePtr RenderUtil::Scene() const
 {
   return this->dataPtr->scene;
 }
+//////////////////////////////////////////////////
+void RenderUtil::UpdateECM(const UpdateInfo &,
+                           EntityComponentManager &_ecm)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->updateMutex);
+
+  auto entityLightsCmdToDelete =
+    std::move(this->dataPtr->entityLightsCmdToDelete);
+  this->dataPtr->entityLightsCmdToDelete.clear();
```

Comment:
no we can't, because there are two clases trying to get this entity, `Scene3D` and `RenderUtil`. The other variables are using the same logic have a look to the `Update` method

Reply:
When I as coding this I find an issue with the rendering, as you mentioned there is no race conditions, but the `LightCmd` is used in two places: 3DScene and in the Sensor class by the camera sensor. 

If we remove the LightCMD in the 3DScene  then the camera sensor will not get the change. For this reason I added the ISystemUpdate in the Sensor class.

Reply:
The `UpdateECM` is called in two classes. 
 - `Sensors`
https://github.com/ignitionrobotics/ign-gazebo/blob/5bafe4326cf18d5e4c886f8ae09e4403d0a422a1/src/systems/sensors/Sensors.cc#L416
 - `Scene3D`
 https://github.com/ignitionrobotics/ign-gazebo/blob/5bafe4326cf18d5e4c886f8ae09e4403d0a422a1/src/gui/plugins/scene3d/Scene3D.cc#L2703

When this method is called we need to `keep` the data for the other class o classes that may require a render update.
There are two variables (unfortunately with the same name, I can change this). 
 - The `entityLightsCmdToDelete` inside the pimpl pattern is used to keep the state for the following step in the loop. With this be asure that the LightCMD is visible for all the classes that require an update.
 - The local variable `entityLightsCmdToDelete` is used to remove older LigthCMD

Reply:
I updated the name of the variables. The local variable now is called ` olderEntitiesLightsCmdToDelete`

Reply:
Thank you for the clarifications. `Scene3D` and `Sensors` are running on separate processes and are dealing with different instances of the ECM. But I can see how this can become a problem if `Sensors` deletes the component before `SceneBroadcaster` has a chance to propagate the component to the client.

Thanks for walking through this!